### PR TITLE
Dilpreet | Test | UI: Integrate API to view all Imports

### PR DESCRIPTION
**Pull request by dilpreetsinghofficial**
(https://app.clickup.com/t/2r66161)

### CODE REVIEW

**Comment by dilpreetsinghofficial** [comment]
Reviewed

**Review by the-prateek-mishra — APPROVED** [review]

**Comment by dilpreetsinghofficial** [comment]
Reviewed

File: apps/web-giddh/src/app/imports/imports.module.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import { TranslateDirectiveModule } from "../theme/translate/translate.directive.module";
+import { ImportsRoutingModule } from "./imports.routing.module";
+import { ImportsComponent } from "./imports/imports.component";
+
+
+@NgModule({
+    declarations: [ImportsComponent],
+    exports: [],
+    imports: [CommonModule, PaginationModule.forRoot(), MatTableModule, ImportsRoutingModule, MatButtonModule, MatDialogModule, HamburgerMenuModule, TranslateDirectiveModule, NoDataModule,
+        GiddhPageLoaderModule, SharedModule]
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/imports/imports.module.ts]: don't import shared module

File: apps/web-giddh/src/app/imports/imports/imports.component.ts [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+     * @param {boolean} [resetPage]
+     * @memberof ImportsComponent
+     */
+    public getImports(resetPage?: boolean): void {
+        if (resetPage) {
+            this.importRequest.page = 1;
+        }
+        this.isLoading = true;
+        this.importsService.getImports(this.importRequest).pipe(takeUntil(this.destroyed$)).subscribe((response) => {
+            // console.log(response);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/imports/imports/imports.component.ts]: remove commented code

File: apps/web-giddh/src/app/imports/imports/imports.component.ts [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+            this.selectedRangeLabel = value.name;
+        }
+        this.hideGiddhDatepicker();
+        if (value && value.startDate && value.endDate) {
+            this.showClearFilter = true;
+            this.selectedDateRange = { startDate: dayjs(value.startDate), endDate: dayjs(value.endDate) };
+            this.selectedDateRangeUi = dayjs(value.startDate).format(GIDDH_NEW_DATE_FORMAT_UI) + " - " + dayjs(value.endDate).format(GIDDH_NEW_DATE_FORMAT_UI);
+            this.fromDate = dayjs(value.startDate).format(GIDDH_DATE_FORMAT);
+            this.toDate = dayjs(value.endDate).format(GIDDH_DATE_FORMAT);
+            this.importRequest.from = null;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/imports/imports/imports.component.ts]: fix from/to

File: apps/web-giddh/src/app/models/api-models/imports.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    public items?: [];
+}
+
+export interface ImportsData {
+    date?: any;
+    efsPath?: any;
+    expireAt?: any;
+    metaData?: any;
+    importFiile?: any;
+    count?: any;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/models/api-models/imports.ts]: correct data type and formatting

File: apps/web-giddh/src/app/models/api-models/imports.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
+export class ImportsRequest {
+    public from?: string;
+    public to?: string;
+    public page?: number;
+    public count?: number;
+    public totalPages?: number;
+    public totalItems?: number;
+    branchUniqueName?: string;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/models/api-models/imports.ts]:
```suggestion
    public branchUniqueName?: string;
```

File: apps/web-giddh/src/app/services/imports.service.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+export class ImportsService {
+
+    /** This will hold the company uniquename */
+    private companyUniqueName: string;
+
+    constructor(private errorHandler: GiddhErrorHandler, private http: HttpWrapperService,
+        private generalService: GeneralService, @Optional() @Inject(ServiceConfig) private config: IServiceConfigArgs) {
+    }
+
+
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/services/imports.service.ts]: remove blank line

File: apps/web-giddh/src/app/services/imports.service.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    public getImports(importsRequest: ImportsRequest): Observable<BaseResponse<ImportsResponse, ImportsRequest>> {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/services/imports.service.ts:27]: add js doc

File: apps/web-giddh/src/assets/locale/imports/en.json [hunk 8]
```diff
@@ -0,0 +1,14 @@
+{
+    "import_date": "Import Date",
+    "by":"By",
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/locale/imports/en.json]: use from common locale

File: apps/web-giddh/src/assets/locale/imports/en.json [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+    "module": "Module",
+    "import_file": "Import File",
+    "count": "Count",
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/locale/imports/en.json]: add in common

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/assets/locale/hi.json [hunk 10]
```diff
@@ -673,5 +673,6 @@
     "app_purchase_bill": "खरीद बिल",
     "app_on": "On",
     "app_off": "Off",
-    "app_preview_formats": "पूर्वावलोकन के लिए ये फ़ाइल प्रारूप समर्थित नहीं हैं (xls, xlsx, doc, CSV)"
+    "app_preview_formats": "पूर्वावलोकन के लिए ये फ़ाइल प्रारूप समर्थित नहीं हैं (xls, xlsx, doc, CSV)",
+    "app_count": "गिनती करना"
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/locale/hi.json]: fix this

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

**Comment by dilpreetsinghofficial** [comment]
DONE

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**